src/main/scala/nl/knaw/dans/easy/license/PlaceholderMapper.scala

```diff
@@ -0,0 +1,198 @@
+    )
+  }
+
+  def currentDateAndTime = new DateTime().toString("YYYY-MM-dd HH:mm:ss")
+
+  def metadataTable(emd: EasyMetadata)(implicit client: FedoraClient): Observable[ju.List[ju.Map[String, String]]] = {
+    emd.getTerms
+      .asScala
+      .toObservable
+      .map(term => (term, emd.getTerm(term).asScala))
+      .filter { case (_, items) => items.nonEmpty }
+      .flatMap { case (term, items) =>
+        val name = metadataNames.getProperty(term.getQualifiedName)
+        val valueObs = if (term.getName == Term.Name.AUDIENCE) formatAudience(emd)
+        else if (term.getName == Term.Name.ACCESSRIGHTS) Observable.just(formatAccessRights(items.head))
+        else Observable.just(items.mkString(", "))
+
+        valueObs.map(value => Map[KeywordMapping, String](MetadataKey -> name, MetadataValue -> value)
+          .map { case (k, v) => (k.keyword, v) }
+          .asJava)
+      }
+      .foldLeft(new ju.ArrayList[ju.Map[String, String]])((list, map) => { list.add(map); list })
+  }
+
+  private def formatAudience(emd: EasyMetadata)(implicit client: FedoraClient): Observable[String] = {
+    // TODO I need a DisciplineCollectionService or something???
+
+    emd.getEmdAudience.getValues.asScala.toObservable
+      .flatMap(sid => queryFedora(sid, "DC")(_.loadXML \\ "title" text))
+      .reduce(_ + "; " + _)
+  }
+
+  private def formatAccessRights(item: MetadataItem): String = {
+    Try(AccessCategory.valueOf(item.toString)) // may throw an IllegalArgumentException
+      .map {
+        // @formatter:off
+        case ANONYMOUS_ACCESS                 => "Anonymous"
+        case OPEN_ACCESS                      => "Open Access"
+        case OPEN_ACCESS_FOR_REGISTERED_USERS => "Open access for registered users"
+        case GROUP_ACCESS                     => "Restricted - 'archaeology' group"
+        case REQUEST_PERMISSION               => "Restricted - request permission"
+        case ACCESS_ELSEWHERE                 => "Elsewhere"
+        case NO_ACCESS                        => "Other"
+        case FREELY_AVAILABLE                 => "Open Access"
+        // @formatter:on
+      }.doOnError(e => log.warn("No available mapping; using acces category value directly"))
+        .getOrElse(item.toString)
+  }
+
+  def filesTable(datasetID: DatasetID)(implicit client: FedoraClient): Observable[ju.List[ju.Map[String, String]]] = {
+    lazy val query = "PREFIX dans: <http://dans.knaw.nl/ontologies/relations#> " +
+      "PREFIX fmodel: <info:fedora/fedora-system:def/model#> " +
+      s"SELECT ?s WHERE {?s dans:isSubordinateTo <info:fedora/$datasetID> . " +
+      "?s fmodel:hasModel <info:fedora/easy-model:EDM1FILE>}"
+
+    val xs = for {
+      filePid <- queryRiSearch(query)
+      path <- queryFedora(filePid, "EASY_FILE_METADATA")(_.loadXML \\ "path" text)
+      cs = FedoraClient.getDatastream(filePid, "EASY_FILE").execute(client).getDatastreamProfile.getDsChecksum
```

Comment:
`queryRiSearch` and `queryFedora` are both wrapping the fedora API in the package object. Why wrap `getDatastreamDissemination` and not wrap `getDatastream`? Why `queryRiSearch` in the object while its only used here?

Now I'm getting at your question about testing and mocking. `EasyStageFileItemSpec` and `SolrDocumentGeneratorSpec` both make advantage of a Fedora trait. The first creates its own mocker, the latter an expects-matcher. Though I created the first, I'm in favour of the latter. The trait reintroduces having to drag its implementation around in stead of the `FedoraClient` instance I earlier suggested to get rid of.